text/0006-architecture-review-process.md

```diff
@@ -0,0 +1,172 @@
+---
+title: Architecture Review Process
+layout: text
+---
+
+**Version**: 0.1
+
+**Warning**: This document is incomplete. We've added it to the repo before it's finished because we would like a couple of trial runs before we say we're happy, and we would like something to refer to during those reviews.
+
+# Architecture Review Process
+
+## Introduction
+
+Architectural Review decisions within the Firefox organization fall into two categories: "Roadmap" and “Design”.
+
+Product Reviews are not within the scope of this process. This process is about architectural change, which should ultimately be in support of product features, but will generally have an effect on multiple product features. Architectural change can be growing new skills or in reducing large-scale technical debt.
+
+The function of a **Roadmap Review** is to decide if a thing should be done. The goal is to bring together a packet of data to inform a management decision to provide resources to make the thing happen. A Roadmap Review should happen early in the process so that build time isn’t wasted on a "No" decision, but so that enough information is available to management.
+
+A **Design Review** is to decide if the thing should be done in a particular way. The goal is to ensure that the problem space is understood well enough that a realistic bug breakdown can be made, that those working on the problem all agree on the challenge, and that work can be prioritized to hit the hardest parts of the problem first.
```

Comment:
> hardest parts…first

I can think of 2 motivations for this: to get early warning if the hard parts prove impossible or to inform schedule projections early (assuming hard things take the longest). Are these the intents? Either way, I wouldn't mind having the motivations made explicit; so often, it's nice to have the "spirit of the law" written down to function as a guide in exceptional times.

Reply:
Agreed. Now:

> ... to hit the hardest parts of the problem first (in order to get early warning if the hard parts prove impossible or to inform schedule projections early).

Reply:
Fixed in b7d27534